### PR TITLE
implement DoubleEndedIterator trait for DimIter

### PR DESCRIPTION
**Pull request by wcshds**
After implementing DoubleEndedIterator trait for DimIter, I can call the.rev() method on DimIter. This is what I need in bidirectional LSTM.

### Checklist

- Confirmed that `run-checks all` script has been executed.

### CODE REVIEW

File: burn-tensor/src/tests/ops/iter_dim.rs [hunk 1]
```diff
@@ -43,4 +43,47 @@ mod test {
...
+
+    #[test]
+    fn test_iter_dim_double_end() {
+        let input = Tensor::<TestBackend, 1, Int>::arange(0..(4 * 6 * 3)).reshape([4, 6, 3]);
+        let mut iter = input.iter_dim(1);
+
+        let ele0 = Data::from([[[0, 1, 2]], [[18, 19, 20]], [[36, 37, 38]], [[54, 55, 56]]]);
+        let ele1 = Data::from([[[3, 4, 5]], [[21, 22, 23]], [[39, 40, 41]], [[57, 58, 59]]]);
+        let ele2 = Data::from([[[6, 7, 8]], [[24, 25, 26]], [[42, 43, 44]], [[60, 61, 62]]]);
+        #[rustfmt::skip]
```
[the remark this answers is not in the document]
Comment (antimora) [reply burn-tensor/src/tests/ops/iter_dim.rs:55]: Yeah we should remove `#[rustfmt::skip]` and let it format. Tests are code too.

**Review by antimora — CHANGES_REQUESTED** [review]
Looks good. Just please remove `#[rustfmt::skip]` and it's ready to be merged.

**Comment by wcshds** [comment]
#[rustfmt::skip] has been removed, but now the formatted code is a little ugly, haha :)

**Review by dcvz — APPROVED** [review]
LGTM 👍

**Review by antimora — APPROVED** [review]

**Pull request merged**